src/MongoDB.Driver/TransactionExecutor.cs

```diff
@@ -26,6 +26,7 @@ internal static class TransactionExecutor
         // constants
         private const string TransientTransactionErrorLabel = "TransientTransactionError";
         private const string UnknownTransactionCommitResultLabel = "UnknownTransactionCommitResult";
+        private const int ExceededTimeLimitErrorCode = 50;
```

Comment:
In principle we could make ServerErrorCode public, but not sure we want to change the public API right now.

I don't know what `shared` is?

Reply:
placing the ServerErrorCode in the `MongoDB.Shared` folder

Reply:
Oh... I thought you meant

    public enum ServerErrorCode { ... }

or

    shared enum ServerErrorCode { ... }

and I didn't know what `shared` meant.

Reply:
Probably the easiest way is to move this class in the `MongoDB.Shared` folder. It'll help us to avoid a "new" unnecessary public class. I cannot imagine a real situation when user will require having access to this class

Reply:
I can imagine an application looking at the numeric error code when catching an exception and deciding what to do about it or how to phrase an error message to report to the user.

Reply:
As we discussed in the slack, I leave it as is now

Reply:
> I can imagine an application looking at the numeric error code
yes, but probably, in this case, we need to provide a class with the full number of errors. I'm not sure that it's a good idea to use this class(in case if the error is presented in this class) and a just error code (in case if the error is not presented in `ServerErrorCode`) in the same time